### ISSUE TITLE
Test different fractions of the training data for tree score evaluation

### ISSUE THREAD

**Issue by ypotdevin** [ypotdevin/dp-gbdt#18]
So far, the tree score (i.e. the score of the ensemble including the new tree candidate) is evaluated on the same dataset chunk used to create the tree in the first place. (see #17)

Advantage:
- small privacy budget requirement (the chunks are disjoint to another and only a small fraction of the complete data)

Disadvantage:
- high potential of overfitting

Alternatives are:
- use a different chunk (like chunk_0 for the tree, chunk_1 for the score, … chunk_n for the tree, chunk_1 for the score) – as far as I understand this would double the privacy budget requirement
- use the complete dataset – as far as I understand this would multiply the privacy budget requirement by the number of invocations of _Algorithm 1_
